[이영섭] Add: filter기능 추가 및 수정 요청 사항 수정

seobie opened the pull request:
## 수정 사항 간략한 한줄 요약
- filter기능 추가 및 수정 요청 사항 수정
		 
## 수정 사항들 자세한 내용
- `filter`기능 추가 구현
- `console.log` 삭제 (Monster.js, CardList.js)
- 기능하지 않던 `componentDidMount()` 삭제

## 기타 질문 및 특이 사항

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

jamessoun93 commented on src/2-monsters/index.js:
```diff
@@ -27,16 +27,37 @@ class App extends Component {
   render() {
     // 필터링 로직
 
+    let filterName = this.state.monsters.filter( monName =>
+       monName.name.toLowerCase().includes(this.state.userInput)
+    )
```
필터링 로직 잘 작성하셨는데 monName.name과 userInput 을 비교할 때 둘다 소문자로 변환을 해주신 후 비교하는게 더 정확합니다.

seobie committed:
Add: Task1~Task4 구현 완료
- Task1 API 호출 구현
- Task2 API 호출의 결과값 props로 자식에게 전달
  - API 호출 후 저장한 배열을 자식 컴포넌트인 CardList.js에 props를
활용하여 넘겨줌
  - 넘겨준 데이터를 CardList.js 에서 console.log로 확인
- Task3 map()사용
  - 넘겨받은 배열을 기준으로 map함수를 활용하여 Card.js 컴포넌트를 리턴
  - Card.js에 id, name, email을 props로 넘겨줌
- Task4 props 활용
  - Card.js 컴포넌트 모양 및 구조 구현

jamessoun93 commented on src/Assignments/Monsters/students/11th/seobie/Components/Card/Card.js:
```diff
@@ -20,13 +20,13 @@ import "./Card.scss";
-          src="https://live.staticflickr.com/4561/38054606355_26429c884f_b.jpg"
+          src={`https://robohash.org/${this.props.dataFromCardList.id}?set=set2&size=180x180`}
           alt=""
         />
-        <h2>Name</h2>
-        <p>Email</p>
+        <h2>{this.props.dataFromCardList.name}</h2>
+        <p>{this.props.dataFromCardList.email}</p>
```
👍👍👍

jamessoun93 commented on src/Assignments/Monsters/students/11th/seobie/Components/CardList/CardList.js:
```diff
@@ -10,8 +10,16 @@ import "./CardList.scss";
 ***********************************************************/
 
 class CardList extends Component {
+  componentDidMount() {}
```
하는 일이 없는 메소드는 지워주세요!

jamessoun93 commented on src/Assignments/Monsters/students/11th/seobie/Components/CardList/CardList.js:
```diff
@@ -10,8 +10,16 @@ import "./CardList.scss";
+  componentDidMount() {}
   render() {
-    return <div className="card-list"></div>;
+    console.log(this.props.monsters);
```
콘솔로그 활용해서 props 로 전달받은 데이터 확인하는 부분 너무 좋습니다! 😉
하지만 콘솔로그 문은 개발단계에서 확인용으로 사용하는 것이기 때문에, push 전에는 꼭 지워주세요~!

jamessoun93 commented on src/Assignments/Monsters/students/11th/seobie/Monsters.js:
```diff
@@ -23,19 +23,32 @@ import "./Monsters.scss";
 
   // 데이터 로딩
+  componentDidMount() {
+    console.log("componentDidMount");
```
여기도 마찬가지입니다!!

jamessoun93 commented on src/Assignments/Monsters/students/11th/seobie/Monsters.js:
```diff
@@ -23,19 +23,32 @@ import "./Monsters.scss";
+      userInput: e.target.value,
+    });
+    console.log(this.state.userInput);
+  };
```
setState 직후에 console.log 로 업데이트 된 state 값을 확인하면 제대로 된 결과값이 나오지 않을텐데요!
왜 그런지 알고 계실까요?! 답변 달아주세요!

seobie replied:
비동기 문제라서 `render` 밑에 찍고 확인해야 하는거로 알고 있습니다!

jamessoun93 replied:
정확해요!! 🥳

jamessoun93 reviewed (changes requested):
리뷰 반영해주세요!

seobie committed:
Add: filter 기능 구현
- filter 기능 구현했습니다.

seobie committed:
Add: filter 기능 추가 및 console.log 삭제
- `filter`기능 추가 구현
- `console.log` 삭제(Monster.js, CardList.js)
- 기능 하지 않던 `componentDidMount()` 삭제

jamessoun93 commented:
LGTM!

수고하셨습니다 🤠
